### ISSUE TITLE
Allow creator of environments to assign access to teams through Pulumi Cloud

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi-cloud-requests#469]
Similar to pulumi/esc#285 (which seems to be when creating using the CLI)

If the environment permissions (https://app.pulumi.com/{org-name}/settings/access-management) are set to None then if a non-organisation admin creates an environment in Pulumi Cloud, there is no way for them to assign access to another team, or even for members of the same team to see the environments that other members of the same team have created.

Only org admins can do this (via the teams page).

tl;dr only org admins can assign ESC environment access to any team, even the one that the creator is on.

**Comment by seanyeh** [comment]
@pierskarsenbarg I tested this locally with the following:
- Created an org and team with 3 members: orgadmin, member1, member2
- member1 created an env env1
- orgadmin sets environment permissions for the org to None.
- Set member1 to team admin (but still not org admin)
- member1 gave team permissions to env1
- member2 can access env1 

So in this case, member1 was not org admin but was able to give env permissions to their team.

Do we want team members who are not admins of the team to be able to grant env permissions also?

**Comment by pierskarsenbarg** [comment]
In your example workflow, if member2 creates an environment, they are the only non-team/org admin person who can see or access it, even though it's created in an organisation. If there was a 4th user (member4) who was in the same team as member3 and created an environment, then member3 wouldn't be able to see it.

The customer expected that it works the same as stacks where if you create a stack you can assign access to your team

**Comment by seanyeh** [comment]
@pierskarsenbarg ok, that makes sense. in investigating this, I accidentally came across what I think is a bug so I created an issue for that as well https://github.com/pulumi/pulumi-service/issues/24157

**Comment by tobiashenkel** [comment]
The actual use case we have is the following. Suppose you have two devops teams `team1` and `team2` with disjoint members and they would like to exchange certain information with each other via esc environments. In this example `team1` is the producer of the data which is to be shared with `team2`:

* One member of `team1` creates an environment `config-for-team2` which contains data that should be consumed by `team2`.
* So `team1` needs to be able to grant (read) access to this environment to `team2` (which no one in `team1` is member of).

The second step is currently not possible because this `grant` operation seems to be actually an operation on the team and not the esc environment.

We already worked around this by employing a naming scheme on our esc environments which contains the "target team" and having a privileged cron job which sets those permissions asynchronously.
